joern-cli/frontends/rubysrc2cpg/src/main/scala/io/joern/rubysrc2cpg/astcreation/AstCreator.scala

```diff
@@ -238,49 +381,53 @@ class AstCreator(filename: String, global: Global)
   }
 
   def astForSymbolContext(ctx: SymbolContext): Ast = {
-    if (ctx.SYMBOL_LITERAL() != null) {
-      val text = ctx.getText
-      val node = NewLiteral()
-        .code(text)
-        .typeFullName(Defines.String)
-        .dynamicTypeHintFullName(List(Defines.String))
-      Ast(node)
-    } else if (ctx.SINGLE_QUOTED_STRING_LITERAL() != null) {
-      Ast()
-    } else {
-      Ast()
-    }
+    val text =
+      if (ctx.SYMBOL_LITERAL() != null) {
+        ctx.getText
+      } else if (ctx.SINGLE_QUOTED_STRING_LITERAL() != null) {
+        ctx.getText
+      } else {
+        return Ast()
+      }
+
+    val node = NewLiteral()
+      .code(text)
+      .typeFullName(Defines.String)
+      .dynamicTypeHintFullName(List(Defines.String))
+    Ast(node)
   }
 
   def astForDefinedMethodNameOrSymbolContext(ctx: DefinedMethodNameOrSymbolContext): Ast = {
     if (ctx == null) return Ast()
 
-    val definedMethodNameCtx = ctx.definedMethodName()
-    val symbolCtx            = ctx.symbol()
-
-    val asts = ListBuffer[Ast]()
-    if (definedMethodNameCtx != null) {
-      asts.addOne(astForDefinedMethodNameContext(definedMethodNameCtx))
-    }
-
-    if (symbolCtx != null) {
-      asts.addOne(astForSymbolContext(symbolCtx))
+    if (ctx.definedMethodName() != null) {
+      astForDefinedMethodNameContext(ctx.definedMethodName())
+    } else {
+      astForSymbolContext(ctx.symbol())
     }
-
-    Ast().withChildren(asts)
   }
 
   def astForAliasStatementContext(ctx: AliasStatementContext): Ast = {
-    if (ctx == null) return Ast()
-
-    val asts = ListBuffer[Ast]()
-    ctx
-      .definedMethodNameOrSymbol()
-      .forEach(dms => {
-        asts.addOne(astForDefinedMethodNameOrSymbolContext(dms))
-      })
-
-    Ast().withChildren(asts)
+    val aliasName = astForDefinedMethodNameOrSymbolContext(
+      ctx
+        .definedMethodNameOrSymbol()
+        .get(0)
+    ).nodes.head
+      .asInstanceOf[NewLiteral]
+      .code
+      .substring(1)
+
+    val methodName = astForDefinedMethodNameOrSymbolContext(
+      ctx
+        .definedMethodNameOrSymbol()
+        .get(1)
+    ).nodes.head
```

Comment:
Use `headOption` to avoid `NoSuchElementException`

Reply:
This place and several other places where I have used `head` are guaranteed to non-null since this is as per that he parser parses. I can use `headOption` instead but that will slightly reduce code readability. 

I do not have a strong opinion for this. Please let me know what you think.